plugins/python-build/bin/python-build

```diff
@@ -876,6 +876,25 @@ build_package_pyston() {
   shopt -u nullglob
 }
 
+build_package_pyston() {
+  # currently supported version 2.3.1 and higher
+  build_package_copy
+  mv "${PREFIX_PATH}/usr/bin" "${PREFIX_PATH}/bin"
+  mv "${PREFIX_PATH}/usr/lib" "${PREFIX_PATH}/lib"
+  local bin
+  shopt -s nullglob
+  for bin in "usr/bin/"*; do
+    if [ -f "${bin}" ] && [ -x "${bin}" ] && [ ! -L "${bin}" ]; then
+      case "${bin##*/}" in
+      "python"* )
+        ( cd "${PREFIX_PATH}/bin" && ln -fs ../usr/bin/"${bin##*/}" "python" )
```

Comment:
Shouldn't it also create a `python3` and `python3.8` aliases in accordance with https://www.python.org/dev/peps/pep-0394/?

```
$ ls -lR ~/.pyenv/versions/pyston-2.3.1/{bin,usr}
/home/vmuser/.pyenv/versions/pyston-2.3.1/bin:
total 8856
-rwxrwxr-x 1 vmuser vmuser     257 сен 24 22:27 pip
-rwxrwxr-x 1 vmuser vmuser     257 сен 24 22:27 pip3
-rwxrwxr-x 1 vmuser vmuser     257 сен 24 22:27 pip3.8
lrwxrwxrwx 1 vmuser vmuser      19 сен 24 22:27 pyston -> python3.8-pyston2.3
lrwxrwxrwx 1 vmuser vmuser      19 сен 24 22:27 pyston3 -> python3.8-pyston2.3
lrwxrwxrwx 1 vmuser vmuser      19 сен 24 22:27 python -> python3.8-pyston2.3
-rwxr-xr-x 1 vmuser vmuser 9051080 сен 24 22:27 python3.8-pyston2.3
-rwxrwxr-x 1 vmuser vmuser     244 сен 24 22:27 wheel
<...>
```

Take a look at `build_package_symlink_version_suffix()` (it's automatically called from `verify_python*`). IIRC you create a `pythonX.Y` executable and it creates the rest for you.

Reply:
I'm not an expert on this stuff but we're generally not providing any symlinks that typically point to a CPython install, such as "python3" or "python3.8", and instead call ourselves "pyston", "pyston3", and "python3.8-pyston2.3". This is more important for things like libraries, where we can't run the compiled modules in `/usr/lib/python3.8`, and need a separate place to put our modules.

Our interpretation is that our version number is "3.8-pyston2.3", so we still adhere to the "python$(VERSION)" convention, except we break anything that assumes version numbers are X.Y.

I'm happy to create a "python3.8" alias as part of the pyenv install if that's what the rest of the code expects, though personally I'd advocate for "python3.8-pyston2.3" getting the same treatment as "python3.8".

Reply:
The point of https://www.python.org/dev/peps/pep-0394/ is so that a script designed for a specific Python version can work with any Python of appropriate version without changes.
So if you don't care about anything not specifically written for Pyston, that indeed doesn't matter to you.

Reply:
I agree that it would be nice if the Pyston pyenv installation included python3 and python3.8 symlinks, but my understanding is that you're advocating that we provide those aliases in our installer. I don't believe that's the right course of action since that is how people refer to the system CPython and I don't think people want that to be redirected to Pyston (though hopefully some day we will get OS distributions to ship Pyston as the system Python). I'm happy to update this PR or the installer code so that it writes out the additional symlinks though.